File: e2e/testdata/node/features/memory_intensive.js
console.log("Starting memory intensive test...");

// 尝试分配大量内存
let largeArray = [];
for (let i = 0; i < 1000000; i++) {
    largeArray.push(i * 2);
    if (i % 100000 === 0) {
        console.log(`Processed ${i} elements, array length: ${largeArray.length}`);
    }
}

// 创建一个大型对象
let largeObject = {};
for (let i = 0; i < 100000; i++) {
    largeObject[`key_${i}`] = `value_${i}`;
}

console.log(`Large object created with ${Object.keys(largeObject).length} properties`);

// 计算一些统计信息
let sum = largeArray.reduce((acc, val) => acc + val, 0);
let average = sum / largeArray.length;

console.log(`Array sum: ${sum}`);
console.log(`Array average: ${average}`);

console.log("Memory intensive test completed successfully");

File: e2e/testdata/python/features/memory_intensive.py
import numpy as np
import sys

print("Starting memory intensive test...")
print(f"Initial memory usage: {sys.getsizeof([])} bytes")

# 尝试分配大量内存
large_list = []
for i in range(1000000):
    large_list.append(i * 2)
    if i % 100000 == 0:
        print(f"Processed {i} elements, current list size: {sys.getsizeof(large_list)} bytes")

# 使用numpy创建大型数组
print("Creating large numpy array...")
large_array = np.random.random((1000, 1000))
print(f"Large array shape: {large_array.shape}, size: {large_array.nbytes} bytes")

# 计算一些统计信息
print(f"Array mean: {large_array.mean()}")
print(f"Array std: {large_array.std()}")

print("Memory intensive test completed successfully")